lightstep/sdk/metric/exporters/otlp/internal/metrictransform/attribute_test.go

```diff
@@ -0,0 +1,270 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrictransform
+
+import (
+	"go.opentelemetry.io/otel/attribute"
+	commonpb "go.opentelemetry.io/proto/otlp/common/v1"
+)
+
+// KeyValue transforms a slice of attribute KeyValues into OTLP key-values.
+func KeyValues(attrs []attribute.KeyValue) []*commonpb.KeyValue {
```

Comment:
Thanks. I just finished updating this PR with complete testing and all the remaining TODOs addressed that I intend to address (of what remains: one was copied from otel-go's otlp/grpc exporter, the other two are histogram overflow corner cases).

Here's what that function became (which confused me as well):

```
func testAttributes(attrs []attribute.KeyValue) []*commonpb.KeyValue {
	// This copy is because attribute.NewSet() sorts in place
	cpy := make([]attribute.KeyValue, len(attrs))
	copy(cpy, attrs)
	return Attributes(attribute.NewSet(cpy...))
}
```